openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -1372,18 +1414,36 @@ def check_handler_compatibility(self,
         #self._long_range_dispersion:'long_range_dispersion'
         #}
 
-    # TODO: nonbondedMethod and nonbondedCutoff should now be specified by StericsForce attributes
     def create_force(self, system, topology, **kwargs):
 
+        self._validate_parameters()
+
+        # If there's no cutoff, make sure the system is nonperiodic/vacuum
+        if (self._long_range_dispersion == "None") and (not topology.is_vacuum):
+            raise SMIRNOFFSpecError("If vdW long_range_dispersion is None, a vacuum/nonperiodic Topology "
+                                    "must be provided")
+        # If there is a cutoff, make sure the system is periodic/condensed
+        if (self._long_range_dispersion != "None") and (not topology.is_condensed):
+            raise SMIRNOFFSpecError("If vdW long_range_dispersion is isotropic or LJPME, a condensed/periodic Topology "
+                                    "must be provided")
+
+
         force = openmm.NonbondedForce()
-        nonbonded_method = self._NONBOND_METHOD_MAP[self._nonbonded_method]
-        force.setNonbondedMethod(nonbonded_method)
-        force.setCutoffDistance(self._cutoff.in_units_of(unit.nanometer))
-        if 'ewaldErrorTolerance' in kwargs:
-            force.setEwaldErrorTolerance(kwargs['ewaldErrorTolerance'])
-        if 'useDispersionCorrection' in kwargs:
-            force.setUseDispersionCorrection(
-                bool(kwargs['useDispersionCorrection']))
+        # If we're using PME, then the only possible openMM Nonbonded type is LJPME
+        if self._long_range_dispersion == 'LJPME':
+            force.setNonbondedMethod(openmm.NonbondedForce.LJPME)
+            force.setCutoffDistance(self._cutoff.in_units_of(unit.nanometer))
+        elif self._long_range_dispersion == 'isotropic':
+            force.setNonbondedMethod(openmm.NonbondedForce.PME)
+            force.setCutoffDistance(self._cutoff.in_units_of(unit.nanometer))
+            if hasattr(self, '_switch_width'):
+                force.setUseSwitchingFunction(False)
+            else:
+                force.setUseSwitchingFunction(True)
+                force.setSwitchingDistance((self._cutoff - self._switch_width).in_units_of(unit.nanometer))
+        elif self._long_range_dispersion == "None":
+            force.setNonbondedMethod(openmm.NonbondedForce.NoCutoff)
```

Comment:
You can use PME and no long-range dispersion. I think the logic here is confused.

Reply:
@jchodera I think I've got it now -- refactored the logic in 0a55e67 if you'd like to check this out again.